### PR TITLE
add discount to products

### PR DESCRIPTION
**Pull request by Guilherme-BS**
# Descrição
add discount on products
I declared a global function that set the discount and then I only called it in the necessary places
## Changes
before:
Captura de tela de 2022-03-07 22-14-17
after :
Captura de tela de 2022-03-07 22-11-25

## Notes
- note one : v0 I intend to implement the discount icon at the top of the image of each product

### CODE REVIEW

File: packages/components/CartItem.tsx [hunk 1]
```diff
@@ -37,6 +37,7 @@ const Text = chakra(ChakraText, {
 });
 
 export default function CartItem({ item, onQuantityChange, onRemove }: Props) {
+  console.log();
```
Comment (Filipe-barbosa) [inline packages/components/CartItem.tsx]: acho que esse console log ficou esquecido aqui hehe

File: packages/components/NavBar.tsx [hunk 2]
```diff
@@ -16,7 +16,7 @@ function NavBar() {
 
   cart.items.forEach((item) => {
     quantity += item.quantity;
-    amount += item.quantity * item.product.price;
+    amount += item.quantity * calculateDiscount(item.product);
```
Comment (Filipe-barbosa) [inline packages/components/NavBar.tsx]: seria legal colocar esse valor em uma variável  ao invés de chamar a funcao no momento do calculo ?

**Review by Filipe-barbosa — COMMENTED** [review]
Bom trabalho.

File: db.ts [hunk 3]
```diff
@@ -6,6 +6,7 @@ export const products: IProduct[] = [
...
     price: 10,
+    discountPrice: 34,
```
Comment (cassios) [inline db.ts]: como o desconto é percentual, acho que poderíamos chamar a propriedade apenas de `discount`

**Review by cassios — COMMENTED** [review]
boaaa, mamãooooooo 👏🏽 👏🏽 👏🏽

File: db.ts (hunk 3 above)
Comment (ThiagoPereiraSje) [reply db.ts]: Não sei se estou certo, mas acho que tentaria resolver dessa forma:

`export default interface IProduct {
  id: number;
  title: string;
  img: string;
  price: number;
  discount: number;
  discountedPrice?: number; // tentaria criar uma computed property
  description: string;
  fullDescription: string;
}

export class Product implements IProduct {
  public id: number;
  public title: string;
  public img: string;
  public price: number;
  public discount: number;
  public description: string;
  public fullDescription: string;

  constructor(product: IProduct) {
    this.id = product.id;
    this.title = product.title;
    this.img = product.img;
    this.price = product.price;
    this.discount = product.discount;
    this.description = product.description;
    this.fullDescription = product.fullDescription;
  }

  get discountedPrice() {
    return this.price - (this.price * this.discount) / 100;
  }
}`

Comment (ThiagoPereiraSje) [reply db.ts]: E como resultado o banco de dados ficaria desse jeito:

`import IProduct, { Product } from "./entities/IProduct";

export const products: IProduct[] = [
  new Product({
    id: 3,
    title: "Camiseta 3",
    img: "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcR6XGdJw5NsYRRklb8YpamARtn8YTG5aOXVhg&usqp=CAU",
    price: 10,
    discount: 10,
    description:
      "It is a long established fact that a reader will be distracted.",
    fullDescription:
      "It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout. The point of using Lorem Ipsum is that it has a more-or-less normal distribution of letters, as opposed to using Content here, content here, making it look like readable English. ",
  }),
  new Product({
    id: 4,
    title: "Camiseta 3",
    img: "https://aramismenswear.vtexassets.com/arquivos/ids/742946-800-auto?v=637525574779270000&width=800&height=auto&aspect=true",
    price: 10,
    discount: 20,
    description:
      "It is a long established fact that a reader will be distracted.",
    fullDescription:
      "It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout. The point of using Lorem Ipsum is that it has a more-or-less normal distribution of letters, as opposed to using Content here, content here, making it look like readable English. ",
  }),
  new Product({
    id: 5,
    title: "Camiseta 5",
    img: "https://d2cnvqfvkdm3ii.cloudfront.net/Custom/Content/Products/16/44/1644118_blusa-le-lis-blanc-devore-i-feminina-11-11-3731_t18_637469137371272089.jpg",
    price: 10,
    discount: 30,
    description:
      "It is a long established fact that a reader will be distracted.",
    fullDescription:
      "It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout. The point of using Lorem Ipsum is that it has a more-or-less normal distribution of letters, as opposed to using Content here, content here, making it look like readable English. ",
  }),
  new Product({
    id: 15,
    title: "Caneta 17",
    img: "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTsxzknPvCjyu0l2mlfxKhmv3xy3YzSPSukLg&usqp=CAU",
    price: 10,
    discount: 0,
    description:
      "It is a long established fact that a reader will be distracted.",
    fullDescription:
      "It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout. The point of using Lorem Ipsum is that it has a more-or-less normal distribution of letters, as opposed to using Content here, content here, making it look like readable English. ",
  }),
];
`

Comment (ThiagoPereiraSje) [reply db.ts]: E acho que mudaria 'discountedPrice' para 'computedPrice'

File: packages/components/NavBar.tsx [hunk 4]
```diff
@@ -15,8 +15,9 @@ function NavBar() {
   let amount = 0;
 
   cart.items.forEach((item) => {
+    const calculateDiscountPrice = calculateDiscount(item.product);
```
Comment (cassios) [inline packages/components/NavBar.tsx]: o resultado de `calculateDiscount(item.product)` é o preço com desconto, certo? Acho q o nome da variável não ta muito bacana pq ele dá a entender que é uma função. O que acha de apenas `discountPrice`? Ou até chamar a função direto na hora de multiplicar?

**Review by cassios — APPROVED** [review]
Boa, Mamão.
Lembrete: juntar commits qdo for mergear. Se precisar de ajuda, me chama

**Review by ThiagoPereiraSje — APPROVED** [review]
Está tudo blz, mas @Guilherme-BS você ainda vai resolver esse problema usando classes ?

**Pull request merged**